spec/test-data/translations/file-with-duplicated-keys.strings

```diff
@@ -56,11 +56,19 @@ Another line.";
 "The Google account \"%@\" doesn't match any account on WordPress.com" = "localized translation here";
 /* A red herring that might be seen as duplicate to the previous one if our RegExp does not correctly account for escaped quotes */
 "The Google account \"%@\" is invalid" = "localized translation here";
-/* Duplicated \"%@\" */
+/* Duplicated keys that include escaped quotes */
 "key with \"%@\" character" = "localized translation here";
 "key with \"%@\" character" = "localized translation here";
+/* Duplicated keys that include escaped quotes, interleaved */
 "key with multiple \"%@\" escapes \":)\" in it" = "localized translation here";
+"Login to a \"%@\" account" = "Login to a \"%@\" account"; // Here %@ = the service name, like Google
 "key with multiple \"%@\" escapes \":)\" in it" = "localized translation here";
-/* This is just a check for Gio's implementation. We might want to remove it? */
+"Login to a \"%@\" account" = "Login to a \"%@\" account"; // Here %@ = the service name, like Google
+/* Duplicated keys with escaped quotes and equals */
+"key with \"%@\" and = character" = "localized translation here";
+"key with \"%@\" and = character" = "localized translation here";
```

Comment:
The idea (from what I talked in my comment) would have been to have a red herring here instead of an actual duplicate, to show that the RegEx would mistakenly see `"key with \"%@\" and =` as the key + equal sign left of the key, and not see the rest of the key (i.e. ` character`)

So I think the goal of the test here should instead be:

```suggestion
"key with \"%@\" and = character for case 1" = "localized translation here";
"key with \"%@\" and = character for case 2" = "localized translation here";
```
And then make the test ensure that those would NOT be considered the same, duplicate keys, but be properly seen as different ones.

Reply:
@mokagio Played around with https://regex101.com/ to find some more red herring cases, and added one of them in 8d4eb9e, which makes the test fail — proving that the current RegEx is not resilient enough 😒 

The more I think about it the more I think a mini-state-machine line parser — which would read one character at a time, notice that it's in an escaping context when it encounters a backslash and eats the following character without interpreting it in such case, and notice when we enter/exit a quoting context… — might be a safer option?

_(As a bonus, such a solution would also support unquoted keys — like we have in `InfoPlist.strings` files; though that action to find duplicates is more likely to be run on files that we get back from GlotPress rather than files manually maintained, and those GlotPress files probably always get their keys quoted during export… 🤷 )_

Some alternatives to going the mini-state-machine route would be:
 - Either to first `gsub('\\', '\u0001').'gsub('\"', '\u0001')` (using `\0001` for its high unlikeliness to be used in real keys) before running the RegEx, thus avoiding it to risk being tripped by escaped quotes, but also double-tripped by "escaped backslash right before proper unescaped quote" (eg `"key ends in backslash char \\"`. And then we can just us a RegEx that will match `^\s*"([^"]+)"`. Not sure how confident we could be of such a hacky solution though…
 - Or alternatively, work even harder in a RegEx that would be able to represent a rule of "any character other than a quote… except if that quote is preceeded by a backslash". But again we might get tripped by "double-backslash before quote" case… so that might make the RegEx quite complicated, unreadable, and hard to trust as well.

Hence the idea of the mini-state-machine starting to feel more and more appealing to me… wdyt?

Reply:
@mokagio Couldn't resist at taking a stab at how a state machine would look 😅  You can see my attempt [in the `duplicate-keys-state-machine` branch](https://github.com/wordpress-mobile/release-toolkit/compare/add-helper-to-lint-for-duplicated-keys-in-strings..duplicate-keys-state-machine) 🙃 

Make what you will of it, build on it or toss it in the trash and ignore it completely if you think that's not the right approach, up to you… but figured I'd share it anyway 😃 

Btw, that attempt made me discover that your example `.strings` fixture had a syntax error — which my state machine detected, while your code couldn't 😛 — where a `;` was missing at the end of line 14 😇

Reply:
@AliSoftware 

> The idea (from what I talked in my comment) would have been to have a red herring here instead of an actual duplicate

Duh 🤦‍♂️ I focused more on the "with `=`" part than on the actual inner workings of the RegEx and how to trick it.

>  Couldn't resist at taking a stab at how a state machine would look 😅 

Oh sweet. Thanks. Given it works better than the current implementation, I think we should use it.

It is more verbose than the substitution approach, but I feel like the substitution is "dirty". It's a neat trick, but the state machine approach seems more solid. You also mention working even harder on the RegEx, but I agree that we might end up shooting ourselves in the foot with that.

Reply:
@AliSoftware I cherry-picked your work, then made it the only implementation via [`f4b5db2`](https://github.com/wordpress-mobile/release-toolkit/pull/342/commits/f4b5db2ece4e90441844a36849e56d86aaebcf6b).

I'll keep working on this next week, my idea is to extract this into the helper as discussed [above](https://github.com/wordpress-mobile/release-toolkit/pull/342#discussion_r816748689). Then, I think I'd call this PR "done" and open a followup to use the new helper in a Fastlane action.